cmd/otelcorecol/main.go

```diff
@@ -11,8 +11,7 @@ import (
 )
 
 func main() {
-	factories, err := components()
-	if err != nil {
+	if _, err := components(); err != nil {
```

Comment:
With `components()` called twice, it's possible the second call will swap out a component the second time around if a feature gate is set and end up error'ing differently at line 25 right? Would it be possible to move the initialization of the feature gates up instead of calling this twice?

Reply:
> With `components()` called twice, it's possible the second call will swap out a component the second time around if a feature gate is set and end up error'ing differently at line 25 right?

I did not consider that. But yes, that might be the case. Making the first call somehow useless.

> Would it be possible to move the initialization of the feature gates up instead of calling this twice?

Defiantly, that is what I tried first. Any recommendations how to do that? What I had in mind was creating another root command and parse only the featuregate flags. This only worked, when I ignored the config flag using https://github.com/spf13/cobra/pull/662.

But since I moved the featureflag evaluation out, the original root command started to fail.